packages/ovm/src/contracts/ExecutionManager.sol

```diff
@@ -494,55 +501,6 @@ contract ExecutionManager is FullStateManager {
         }
     }
 
-    /********* Utils *********/
-
-    /**
-     * @notice Initialize a new context, setting the timestamp, queue origin, and gasLimit as well as zeroing out the
-     *         msgSender of the previous context.
-     *         NOTE: this zeroing may not technically be needed as the context should always end up as zero at the end of each execution.
-     * @param _timestamp The timestamp which should be used for this context.
-     * @param _queueOrigin The queue which this context's transaction was sent from.
-     * @param _ovmTxOrigin The tx.origin for the currently executing transaction. It will be ZERO_ADDRESS if it's not an EOA call.
-     */
-    function initializeContext(uint _timestamp, uint _queueOrigin, address _ovmTxOrigin) internal {
-        // First zero out the context for good measure (Note ZERO_ADDRESS is reserved for the genesis contract & initial msgSender)
-        restoreContractContext(ZERO_ADDRESS, ZERO_ADDRESS);
-        // And finally set the timestamp, queue origin, & tx origin
-        executionContext.timestamp = _timestamp;
-        executionContext.queueOrigin = _queueOrigin;
-        executionContext.ovmTxOrigin = _ovmTxOrigin;
-    }
-
-    /**
-     * @notice Change the active contract to be something new. This is used when a new contract is called.
-     * @param _newActiveContract The new active contract
-     * @return The old msgSender and activeContract. This will be used when we restore the old active contract.
-     */
-    function switchActiveContract(address _newActiveContract) internal returns(address _oldMsgSender, address _oldActiveContract) {
-        // Store references to the old context
-        _oldActiveContract = executionContext.ovmActiveContract;
-        _oldMsgSender = executionContext.ovmMsgSender;
-        // Set our new context
-        executionContext.ovmActiveContract = _newActiveContract;
-        executionContext.ovmMsgSender = _oldActiveContract;
-        // Emit an event so we can track the active contract. This is used in order to parse transaction receipts in the fullnode
-        emit ActiveContract(_newActiveContract);
-        // Return old context so we can later revert to it
-        return (_oldMsgSender, _oldActiveContract);
-    }
-
-    /**
-     * @notice Restore the contract context to some old values.
-     * @param _msgSender The msgSender to be restored.
-     * @param _activeContract The activeContract to be restored.
-     */
-    function restoreContractContext(address _msgSender, address _activeContract) internal {
-        // Revert back to the old context
-        executionContext.ovmActiveContract = _activeContract;
-        executionContext.ovmMsgSender = _msgSender;
-    }
-
-
```

Comment:
Just note that this massive deletion & then the addition below is actually the result of a failed diff. I just changed the the `Utils` comment at the top, added a param to `initializeContext(...)` & then added the function `getL1MessageSender()` at the bottom.